client/app/bundles/course/assessment/submission/components/ScribingView/ScribingCanvas.jsx

```diff
@@ -0,0 +1,549 @@
+      left: scribble.left * factor,
+      top: scribble.top * factor,
+    });
+  }
+
+  denormaliseScribble(scribble) {
+    return this.normaliseScribble(scribble, true);
+  }
+
+  initializeScribbles = (isFirstInit) => {
+    const { scribbles } = this.props.scribing.answer;
+    const userId = this.props.scribing.answer.user_id;
+
+    this.isScribblesLoaded = false;
+    let userScribble = [];
+
+    if (isFirstInit) {
+      this.canvas.layers = [];
+    } else {
+      this.canvas.layers.forEach(layer => this.canvas.add(layer.scribbleGroup));
+    }
+
+    if (scribbles) {
+      scribbles.forEach((scribble) => {
+        const objects = JSON.parse(scribble.content).objects;
+        const fabricObjs = [];
+
+        // Parse JSON to Fabric.js objects
+        for (let i = 0; i < objects.length; i++) {
+          // eslint-disable-next-line no-undef
+          const klass = fabric.util.getKlass(objects[i].type);
+          switch (objects[i].type) {
+            case 'path': {
+              klass.fromObject(objects[i], (obj) => {
+                this.denormaliseScribble(obj);
+                fabricObjs.push(obj);
+              });
+              break;
+            }
+            case 'i-text':
+            case 'line':
+            case 'rect':
+            case 'ellipse': {
+              const obj = klass.fromObject(objects[i]);
+              this.denormaliseScribble(obj);
+              fabricObjs.push(obj);
+              break;
+            }
+            default: {
+              break;
+            }
+          }
+        }
+
+        // Create layer for each user's scribble
+
+        // Layer for other users' scribble
+        // Disables scribble selection
+        if (isFirstInit && scribble.creator_id !== userId) {
+          // eslint-disable-next-line no-undef
+          const scribbleGroup = new fabric.Group(fabricObjs);
+          scribbleGroup.selectable = false;
+
+          const showLayer = (isShown) => {
+            // eslint-disable-next-line no-param-reassign
+            scribbleGroup._objects.forEach(obj => (obj.visible = isShown));
+          };
+          // Populate layers list
+          const newScribble = {
+            ...scribble,
+            isDisplayed: true,
+            showLayer,
+            scribbleGroup,
+          };
+          this.canvas.layers = [...this.canvas.layers, newScribble];
+          this.canvas.add(scribbleGroup);
+        } else {
+          // Add other user's layers first to avoid blocking of user's layer
+          userScribble = fabricObjs;
+        }
+      });
+
+      // Layer for current user's scribble
+      // Enables scribble selection
+      userScribble.map(obj => (this.canvas.add(obj)));
+    }
+    this.canvas.renderAll();
+    this.isScribblesLoaded = true;
+  }
+
+  initializeCanvas(answerId, imagePath) {
+    const imageUrl = `https://coursemology.org/${imagePath}`;
+    // const imageUrl = `${window.location.origin}/${imagePath}`;
```

Comment:
I was trying to use the images deployed at coursemology.org for testing in localhost, forgot to change it back 😅